### DISCUSSION TITLE
Fetching the same data in multiple components

### DISCUSSION THREAD

**Discussion by LuckyOneDev** [pmndrs/zustand#2410] [Q&A]
So here is the problem. I have data that should be accessible in multiple components. I fetch data and use it like this:

```
const userInfo = useUserStore((s) => s.userInfos[0]);
const fetchUserInfo = useUserStore((s) => s.fetch);
useEffect(() => {
	fetchUserInfo(0); // 0 is user id
}, []);
```

But calling fetchUserInfo in **every** component that needs userInfo seems dirty. I would like a syntax like this:
```
const userInfo = useUserStore((s) => s.userInfos[0]);
// If data already loaded userInfo contains it. Otherwise it initiates request to load it and becomes null until data loads.
```

I already have deduplicated actual request to the server but using useEffect feels just wrong.

Data is parametrised so it is impossible to fetch before component load

**Comment by dbritto-dev** [comment]
@LuckyOneDev this example lack of some parts, would you mind sharing us a minimal repro on stackblitz?
